### PR TITLE
Add ARM IntelliSenseModes

### PR DESCRIPTION
**Pull request by michelleangela**
Add the following ARM IntelliSenseModes

IntelliSenseMode | description
-- | --
clang-arm | Targets ARM 32-bit   architecture and simulates compiling with Clang
clang-arm64 | Targets ARM 64-bit   architecture and simulates compiling with Clang
gcc-arm | Targets ARM 32-bit   architecture and   simulates   compiling with Gcc
gcc-arm64 | Targets ARM 64-bit   architecture and   simulates   compiling with Gcc
msvc-arm | Targets ARM 32-bit   architecture and simulates compiling with msvc (Microsoft C++ compiler)
msvc-arm64 | Targets ARM 64-bit   architecture and simulates compiling with msvc (Microsoft C++ compiler)

Feature request: https://github.com/microsoft/vscode-cpptools/issues/4271

### CODE REVIEW

File: Extension/bin/msvc.arm64.msvc.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "defaults": [
+    "-D_MSC_EXTENSIONS",
+    "--microsoft",
+    "--microsoft_bugs",
+    "--microsoft_version",
+    "1924",
```
Comment (sean-mcmanus) [inline Extension/bin/msvc.arm64.msvc.json]: 1926

File: Extension/bin/msvc.arm32.msvc.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "defaults": [
+    "-D_MSC_EXTENSIONS",
+    "--microsoft",
+    "--microsoft_bugs",
+    "--microsoft_version",
+    "1924",
```
Comment (sean-mcmanus) [inline Extension/bin/msvc.arm32.msvc.json]: 1926

File: Extension/bin/msvc.x86.msvc.json [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
-      "-D_M_IX86_FP=2"
-    ],
-    "defaults_op": "merge"
+  {
+    "defaults": [
+      "-D_MSC_EXTENSIONS",
+      "--microsoft",
+      "--microsoft_bugs",
+      "--microsoft_version",
+      "1924",
```
Comment (sean-mcmanus) [inline Extension/bin/msvc.x86.msvc.json]: 1926

File: Extension/bin/msvc.x64.msvc.json [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
-      "-D_M_AMD64=100"
-    ],
-    "defaults_op": "merge"
+  {
+    "defaults": [
+      "-D_MSC_EXTENSIONS",
+      "--microsoft",
+      "--microsoft_bugs",
+      "--microsoft_version",
+      "1924",
```
Comment (sean-mcmanus) [inline Extension/bin/msvc.x64.msvc.json]: 1926

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]

**Review by sean-mcmanus — APPROVED** [review]

**Comment by aelray** [comment]
I'm using 0.28.0-insiders3 and it doesn't work with ARMClang. I get incorrect code highlighting until I add some defines, which usually the compiler would provide. Also includes are handled wrongly, it still tries to resolve stdint.h from the c stdlib instead of from the ARMClang toolchain directory.

**Comment by michelleangela** [comment]
@aelray, could you create a new issue and provide your c_cpp_properties.json and the [C/C++ language server logging](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server) when opening the project? The defines should have been queried from the compiler. Perhaps the extension failed to query the compiler so it could not get the defines and relative paths of the library header files.

**Pull request merged**